src/VisualStudio/Core/Def/Implementation/VirtualMemoryNotificationListener.cs

```diff
@@ -61,6 +68,17 @@ public int OnBroadcastMessage(uint msg, IntPtr wParam, IntPtr lParam)
                         }
 
                         _workspaceCacheService.FlushCaches();
+
+                        // turn off full solution analysis
+                        if (_workspace.Options.GetOption(RuntimeOptions.FullSolutionAnalysis))
+                        {
+                            _workspace.Services.GetService<IOptionService>().SetOptions(_workspace.Options.WithChangedOption(RuntimeOptions.FullSolutionAnalysis, false));
+
+                            // let user know full analysis is turned off due to memory concern
+                            // no close info bar action
+                            _workspace.Services.GetService<IErrorReportingService>().ShowErrorInfo(ServicesVSResources.FullSolutionAnalysisOff, () => { });
```

Comment:
I thought about that before and choose not to do it since they will certainly OOM. this option change will let diagnostic service to throw away any data we no longer needed.

the reason I gave users an option to turn this back on is if user wants to working on other solution, they can go ahead without restarting VS by set those options back on.